buildSrc/src/main/java/com/r3/csde/BuildCPIsHelper.java

```diff
@@ -0,0 +1,270 @@
+        Process proc = pb.start();
+        proc.waitFor();
+
+    }
+
+    public void buildCPIs() throws IOException, InterruptedException, CsdeException {
+        createCorDappCPI();
+        createNotaryCPI();
+    }
+
+    private void createCorDappCPI() throws IOException, InterruptedException, CsdeException {
+
+        String appCPIFilePath = pc.workflowBuildDir + "/" +
+                pc.project.getRootProject().getName() + "-" +
+                pc.project.getVersion() + ".cpi";
+
+        File appCPIFile = new File(appCPIFilePath);
+        appCPIFile.delete();
+
+        File srcDir = new File(pc.workflowBuildDir + "/libs");
+        File[] appCPBs = srcDir.listFiles(( x , name ) -> name.endsWith(".cpb"));
+        if (appCPBs == null) throw new CsdeException("Expecting exactly one CPB but no CPB found.");
+        if (appCPBs.length != 1) throw new CsdeException("Expecting exactly one CPB but more than one found.");
+
+        pc.out.println("appCpbs:");
+        pc.out.println(appCPBs[0].getAbsolutePath());
+
+        LinkedList<String> commandList = new LinkedList<>();
+
+        commandList.add(String.format("%s/java", pc.javaBinDir));
+        commandList.add(String.format("-Dpf4j.pluginsDir=%s/plugins/", pc.cordaCliBinDir));
+        commandList.add("-jar");
+        commandList.add(String.format("%s/corda-cli.jar", pc.cordaCliBinDir));
+        commandList.add("package");
+        commandList.add("create-cpi");
+        commandList.add("--cpb");
+        commandList.add(appCPBs[0].getAbsolutePath());
+        commandList.add("--group-policy");
+        commandList.add(pc.devEnvWorkspace + "/GroupPolicy.json");
+        commandList.add("--cpi-name");
+        commandList.add(pc.appCPIName);
+        commandList.add("--cpi-version");
+        commandList.add(pc.project.getVersion().toString());
+        commandList.add("--file");
+        commandList.add(appCPIFilePath);
+        commandList.add("--keystore");
+        commandList.add(pc.devEnvWorkspace + "/signingkeys.pfx");
+        commandList.add("--storepass");
+        commandList.add("keystore password");
+        commandList.add("--key");
+        commandList.add("my-signing-key"); // todo: should be passed as context property
+
+        ProcessBuilder pb = new ProcessBuilder(commandList);
+        pb.redirectErrorStream(true);
+        Process proc = pb.start();
+        proc.waitFor();
+
+// todo: work out how to capture error code better than the following code
+
+//        BufferedReader reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+//        File tempOutputFile = new File(String.format("%s/tempOutput.txt", pc.devEnvWorkspace));
+//        tempOutputFile.delete();
+//        FileWriter fileWriter = new FileWriter(tempOutputFile);
+//        String line;
+//        while (( line = reader.readLine()) != null){
+//            fileWriter.write(line + "\n");
+//        }
+//        fileWriter.close();
+
```

Comment:
Do we still need this commented out code? Was it moved elsewhere?

Reply:
I need that for a future bit of refactoring